up-l1/README.adoc

```diff
@@ -50,157 +42,218 @@ We will use the following terminology during this document:
+| link:../basics/uri.adoc[UUri]
+| The _source_ address pattern that messages need to match.
 
+| sinkFilter
+| link:../basics/uri.adoc[UUri]
+| The _sink_ address pattern that messages need to match. If omitted, a message must not contain any sink address in order to match.
+
+| listener
 | <<UListener>>
-| Listener to be registered
+| The listener to be registered.
 
+| result
+| Future<Void, link:../basics/error_model.adoc[UStatus]>
+| A succeeded future if the listener has been registered successfully. +
+Otherwise, a failed future containing a `UStatus` indicating the reason for the failure.
 |===
 
+[oft-sid="req~utransport-register-listener~1"]
+--
+Implementations
 
-=== UnregisterListener()
+* *MUST* support registering more than one listener for any given address patterns
+* *MUST* support registering the same listener for multiple address patterns
+* *MUST* document the maximum supported number of listeners per address pattern.
+* *MUST* fail with a `UCode::RESOURCE_EXHAUSTED`, if the maximum number of listeners is reached
+--
 
-API used to unregister a <<UListener>> for a given topic.
+=== UnregisterListener
 
-API Signature: 
+Clients use this method to unregister a previously registered listener.
+The listener will no longer be called for any (matching) messages after this method has returned successfully.
 
-`OUT <<UStatus>> unregisterListener(IN UUri, IN <<UListener>>&)`
+[source]
+----
+unregisterListener(
+  IN sourceFilter: UUri,
+  IN sinkFilter: UUri?,
+  IN listener: UListener&,
+  OUT result: Future<Void, UStatus>
+)
+----
 
-==== Parameters
-.UnregisterListener Parameters
-[width="70%",cols="20%,80%"]
+.RegisterListener Parameters
+[width="100%",cols="15%,25%,60%"]
 |===
-|Parameter | Description
+|Parameter | Type | Description
 
-| UUri
-| Topic to unregister the listener for
+| sourceFilter
+| link:../basics/uri.adoc[UUri]
+| The source address pattern that the listener had been registered for.
 
+| sinkFilter
+| link:../basics/uri.adoc[UUri]
+| The sink address pattern that the listener had been registered for.
+
+| listener
 | <<UListener>>
-| Listener to be unregistered
+| The listener to be unregistered.
 
+| result
+| Future<Void, link:../basics/error_model.adoc[UStatus]>
+| A succeeded future if the listener has been unregistered successfully. +
+Otherwise, a failed future containing a `UStatus` indicating the reason for the failure.
 |===
 
-* *MUST* pass a UUri for which `isMicroForm()` returns true
-* *SHOULD* pass a Resolved UUri (contains both names and numbers)
-
 
-=== Receive()
+=== Receive
 
-Implements the _pull_ <<Delivery Method>> to fetch a message from the transport for a given UUri (topic).
+Clients use this method to receive a single message matching given filter criteria.
+This method implements the _pull_ <<Delivery Method>> for the underlying messaging infrastructure.
 
-`OUT <<UStatus>> receive(IN UUri, OUT UPayload&, OUT UAttributes&)`
+[source]
+----
+receive(
+  IN sourceFilter: UUri,
```

Comment:
Is the new line for each attribute like this a standard for defining APIs?

Reply:
I just thought that it would improve readability because you could at once see which are the inputs and the outputs.